pangeo_forge_recipes/storage.py

```diff
@@ -37,10 +37,13 @@ def _copy_btw_filesystems(input_opener, output_opener, BLOCK_SIZE=10_000_000):
     with input_opener as source:
         with output_opener as target:
             while True:
-                data = source.read(BLOCK_SIZE)
-                if not data:
-                    break
-                target.write(data)
+                try:
+                    data = source.read(BLOCK_SIZE)
+                    if not data:
+                        break
+                    target.write(data)
+                except ValueError as e:
+                    raise BlockSizeError() from e
```

Comment:
I'm concerned that catching a ValueError here is not necessarily going to be a reliable test for the block size error. The reason is that there are many _other_ things that might go wrong and raise a `ValueError` within the `try` block that are totally unrelated to block size. For example, a broken internet connection, invalid write credentials, etc. etc. 

What would be ideal would be if fsspec itself could raise a custom error type for this situation. An alternative would be to parse the error message in `e` and look for the `"Got more bytes"` string reported in https://github.com/pangeo-forge/staged-recipes/pull/31#issuecomment-841522873.

Reply:
I like that parsing `e` idea. (And actually hadn't realized that was possible.) I'll look into it.

Reply:
It would be totally reasonable to have a specific fsspec error for this case.

Reply:
PR submitted https://github.com/intake/filesystem_spec/pull/646